### DISCUSSION TITLE
Thanks for Setting up the repo

### DISCUSSION THREAD

**Discussion by Hello-dot-Jpg** [nk64/GT4286#1] [General]
Awesome work on fixing the button issue.
Here are some random bits of info about the device
The god awful menu system was created using [ysubg](https://github.com/MiyooCFW/gmenu2x)

The hardware may support one of these images https://archive.org/details/@potato_resurrection_project

The Games in the menu are stored in an SQLite data base https://sqlitebrowser.org/dl/
game.db

Updating this should allow you to get the full-length names you wanted I think this would also allow you to move the games from the downloads directory 

The device seems to have active serial, I'll see if I can hook something up and dump the internal eeprom via uboot

**Comment by Hello-dot-Jpg** [comment]
Oh also, there appears to be a dreamcast bios in the bios directory which makes me think that with the right emulator you'd be able to play sega dreamcast games as well 

\bios\dc

**Comment by Hello-dot-Jpg** [comment]
Ok going down the gmenu2x rabbit hole I discovered this
https://github.com/TriForceX/MiyooCFW

I'll investigate further but this may be compatible

**Comment by nk64** [comment]
Hey, thanks for reaching out.

I had hoped that this thing was some sort of Pandora clone, but now I don't think so, on the other hand I don't have enough experience to rule it out for sure, so please let me know if you have any luck copying emulator binaries from another product.

I have updated the readme with a link to a hardware page with the CPU details (though there is not much info around) and pictures of the chips,

I can see that the emulators and the menu (gmenu2x) have all been compiled using a Software Developer Kit (SDK) called [Flythings](https://developer.flythings.cn/en/docs_brief.html). I haven't looked much further than the introduction.

Regarding Sega Dreamcast, I saw vestiges of files left over from a Dreamcast Emulator called [reicast](https://github.com/skmp/reicast-emulator) so I think they might have tried but failed to make something of it.

I have been through the sqlite game.db file like a dose of salts and bent it to my will. So I *do* have nice human readable names for my arcade roms. And I have stopped using the download directory which fixes the issue of save game files polluting the game list.

It is just that the code for that utility is not ready for consumption, but I hope to get that into some sort of command line utility before the new year.

I am primarily a C# developer (and more recently Python for work). I am not so much a Linux, C/C++ SDL / Cross Compiling developer so I think that it is highly unlikely that I will ever get to the stage of compiling a different emulator for it.

**Reply by nk64** [reply]
Please let me know how you identified the active serial pins? I couldn't find a datasheet for the CPU nor obvious serial headers on the board (maybe I didn't look hard enough). My plan (failing finding serial ports and access to uboot) was to desolder the Vcc pin of the Flash chip and dump it with an Arduino SPI reader program or a Buspirate (this is why I bought another one today).

Though we can already run linux commands to copy all the OS files onto the SD card.

**Comment by Hello-dot-Jpg** [comment]
Hi nk64,

I'm a SQA that's worked a lot in the embedded space, I've worked with a few embedded android devises in the past so hopefully I can lend some of my experience to help out although you seem to know you way around pretty well already.

This seems like it would be a fun project to work on and I am sure we would be able to improve upon what they delivered 

I'm not entirely sure how hard it would be to cross compile, but I think if we have the correct toolchain it should hopefully not be too difficult (I might be a little too hopeful on that :D )

> I had hoped that this thing was some sort of Pandora clone, but now I don't think so, on the other hand I don't have enough experience to rule it out for sure, so please let me know if you have any luck copying emulator binaries from another product.

So had I but once you posted the photos of the internals, I realized that it's not, from what I can tell it seems to have originally been used in a handheld device then was used in a "TV Stick" before finally making its way to the arcade console form. Because of it's handheld origins I wonder if the CPU could be overclocked if some sort of active coolng/heatsinking were added.

I've also worked quite a bit with Chinese hardware vendors, and I can say I wouldn't be too concerned about modifying/uploading changes as I suspect this would be a copy of a copy of a copy of an open-source library and would almost certainly be in violation of the licensing with it being used for commercial use.

The only thing I would not host is the ROMs 

>Please let me know how you identified the active serial pins? I couldn't find a datasheet for the CPU nor obvious serial headers on the board (maybe I didn't look hard enough). My plan (failing finding serial ports and access to uboot) was to desolder the Vcc pin of the Flash chip and dump it with an Arduino SPI reader program or a Buspirate (this is why I bought another one today).

I am sure I saw them last night, but I was pretty tired (I have a very active toddler)
I'll have another look, but this guy provides the full pin out of the processor worst comes to worst 
https://linux-chenxing.org/infinity2/

>Though we can already run linux commands to copy all the OS files onto the SD card.

If we can dump the disk images, we should be able to modify the contents of the images and reflash them using uBoot as any changes to the ramdisk partitions would be lost between boots/it would allow us to modify read only partitions

**Comment by Ryhawkian** [comment]
I'm not adding much, but just wanted to say thanks for this. spent all day trying to work out how to screw around with it, and it seems you already have :)

**Reply by nk64** [reply]
Screwing around with it has been more fun that playing the games for me. Let us know if you come across anything interesting.

**Reply by Logicalprocedure** [reply]
I'd also like to add my thanks.
